File: solutions/dart/armstrong-numbers/1/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  // Put your code here
  bool isArmstrongNumber(String numbers) {
    List<String> numbersList = numbers.split('');
    int sum = 0;
    int numberLength = numbersList.length;
    numbersList.forEach((element) {
      int number = int.parse(element);

      num numberPowered = pow(number, numberLength);
      sum += numberPowered as int;
    });
    return sum == int.parse(numbers);
  }
}

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
//        throw new UnsupportedOperationException("Delete this statement and write your own implementation.");
        String[] numbers = String.valueOf(numberToCheck).split("");
        if (numbers.length > 1) {
            int sum = 0;
            for (String number : numbers) {
                sum += (int) Math.pow(Integer.parseInt(number), numbers.length);
            }
            return sum == numberToCheck;
        }
        return true;
    }

}
